### PR TITLE
Level generation

### PR DESCRIPTION
**Pull request by marcelkuipers**
Level pieces are now generated and deleted automatically.
A fly cam, which can be switched to by pressing c is also implemented.

### CODE REVIEW

**Comment by LMiljak** [comment]
The fly cam and level generation are two very separate features and should have been separate branches. So that's for a next time :)

**Comment by NilsHullegien** [comment]
The flycam was added into this branch for debugging of this feature.

**Comment by dVoorhout** [comment]
Maybe implementing a Enum for all the Boolean states would have been cleaner, but because of the way they are used it doesn't matter too much. Everything else look fine

File: ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java [hunk 1]
```diff
@@ -131,7 +209,8 @@ public MainMenu getMainMenu() {
     public Environment getEnv() {
         return environmentState;
     }
-    
+
+    @Override
     public Node getRootNode() {
         return rootNode;
     }
```
Comment (NilsHullegien) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/Main.java:219]: Missing Javadoc

File: ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java [hunk 2]
```diff
@@ -191,8 +263,35 @@ public boolean isInitialized() {
     @Override
     public void setEnabled(boolean arg0) {
         // TODO Auto-generated method stub
+
+    }
+
+    private void updateTestWorld() {
```
Comment (NilsHullegien) [inline ContextDesktop/src/main/java/com/github/migi_1/Context/model/Environment.java]: Missing javadoc

**Comment by NilsHullegien** [comment]
javadoc is fixed now, can someone please check this again (I worked on this branch as well, so I can't do code review)

**Comment by ThatFlint** [comment]
Looks good

**Pull request merged**